### PR TITLE
bamboo variable update

### PR DESCRIPTION
**Pull request by Jazz405**
Updates the common Bamboo variable that Cypress listens for
Close #4895
Supports https://github.com/cypress-io/cypress/pull/6158

### CODE REVIEW

File: source/guides/guides/parallelization.md [hunk 1]
```diff
@@ -236,7 +236,7 @@ Cypress currently uses the following CI environment variables to determine a CI
 Provider  | Environment Variable
 --|--
 AppVeyor  | `APPVEYOR_BUILD_NUMBER`
-Bamboo  | `BAMBOO_BUILD_NUMBER`
+Bamboo  | `bamboo_buildNumber`
```
Comment (jennifer-shehane) [inline source/guides/guides/parallelization.md]: It seems we actually look for `bamboo.buildNumber` https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/util/ci_provider.js#L81:L81

```suggestion
Bamboo  | `bamboo.buildNumber`
```

**Comment by jennifer-shehane** [comment]
I see this is a change required from the https://github.com/cypress-io/cypress/pull/6158 PR now. Please make sure to link to the relevant PR next time so that I have all the full information.

**Comment by jennifer-shehane** [comment]
Changing base branch to 3.8.3-release since that is what this change is targeted for.

**Comment by Jazz405** [comment]
@jennifer-shehane Unable to merge this, did I miss something somewhere?

**Comment by jennifer-shehane** [comment]
Outside contributors cannot merge PRs, it requires a Cypress team member.

**Pull request merged**